### DISCUSSION TITLE
What are best practice using ArcadeDB for multitenant app?

### DISCUSSION THREAD

**Discussion by maw2be** [ArcadeData/arcadedb#1469] [General]
I'm building multitenat app. I was looking for good app to handle this as data model will be complicated, so deciaded to go into graph db, but I also need some RMDB/document features so will give a try with Arcade. And now is my question. How to best handle this? What will be best approach? For regulatory reasons probably will need to have as much seperated data as possible, so I think seperated db. One of my question is how to automate creation of vertex/documents? More questions to come....

**Comment by lvca** [comment]
In most of the cases having one database per tenant is the best choice. ArcadeDB is very efficient with resources and one server can handle hundreds and more of databases without breaking a sweat.

The only con is analytics cross-tenant: in this case, you have to execute the same query against all the databases and group the results at the application level because there is no such thing as a multi-database query (so far).
